src/test/java/calculator/StringAddCalculatorTest.java

```diff
@@ -0,0 +1,46 @@
+package calculator;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class StringAddCalculatorTest {
+    @Test
+    void 빈문자열이거나_null일_경우() {
+        // 불필요한 로컬변수 미생성
+        assertThat(StringAddCalculator.splitAndSum(null)).isEqualTo(0);
+        assertThat(StringAddCalculator.splitAndSum("")).isEqualTo(0);
+    }
+
+    @Test
+    void 쉼표_구분자() {
+        assertThat(StringAddCalculator.splitAndSum("1,2")).isEqualTo(3);
+    }
+
+    @Test
+    void 콜론_구분자() {
+        assertThat(StringAddCalculator.splitAndSum("1,2:3")).isEqualTo(6);
+    }
+
+    @Test
+    void 커스텀_구분자() {
+        assertThat(StringAddCalculator.splitAndSum("//;\n1;2;3")).isEqualTo(6);
+    }
+
+    @Test
+    void 음수_기본구분자() {
+        assertThatThrownBy(() -> {
+            assertThat(StringAddCalculator.splitAndSum("1:-2,3")).isEqualTo(2);
```

Comment:
여기에서는 추가로 검증을 하지 않고 예외가 발생하는 코드만 작성해도 충분할 것 같습니다.
```suggestion
            StringAddCalculator.splitAndSum("1:-2,3");
```